src/gorilla.c

```diff
@@ -115,19 +115,26 @@
 
 // 2^bit
 static inline u_int64_t BIT(u_int64_t bit) {
-    __uint128_t tmp = 1;
-    return (__uint64_t)(tmp << bit);
+    if (__builtin_expect(bit > 63, 0)) {
+        return 0ULL;
+    }
+    return (1ULL << bit);
 }
 
 // the LSB `bits` turned on
 static inline u_int64_t MASK(u_int64_t bits) {
-    __uint128_t tmp = 1;
-    return (__uint64_t)((tmp << bits) - 1);
+    if (__builtin_expect(bits > 63, 0)) {
+        return -1ULL;
+    }
+    return (1ULL << bits) - 1;
 }
 
 // Clear most significant bits from position `bits`
 static inline u_int64_t LSB(u_int64_t x, u_int64_t bits) {
-    return x & MASK(bits);
+    if (__builtin_expect(bits > 63, 0)) {
```

Comment:
@ashtul why do you need this change? isn't leaving it as is will get you the same result?